File: packages/dashboard/src/services/health.ts
import { apiClient } from './api';

export interface HealthStatus {
  status: string;
  timestamp: number;
  version: string;
  service: string;
}

export interface DetailedStatus extends HealthStatus {
  uptime: number;
  memory: {
    used: string;
    available: string;
  };
  connections: {
    active: number;
    total: number;
  };
}

export class HealthService {
  async checkHealth(): Promise<HealthStatus | null> {
    const result = await apiClient.get<HealthStatus>('/api/health');
    if (result.error) {
      console.error('Health check failed:', result.error);
      return null;
    }
    return result.data;
  }

  async getDetailedStatus(): Promise<DetailedStatus | null> {
    const result = await apiClient.get<DetailedStatus>('/api/status');
    if (result.error) {
      console.error('Status check failed:', result.error);
      return null;
    }
    return result.data;
  }
}

export const healthService = new HealthService();

File: packages/cli/src/api/health.rs
use axum::{extract::Extension, response::Result, Json};
use serde_json::{json, Value};
use std::time::{SystemTime, UNIX_EPOCH};

use crate::middleware::CsrfLayer;

pub async fn health_check() -> Result<Json<Value>> {
    let timestamp = SystemTime::now()
        .duration_since(UNIX_EPOCH)
        .map(|d| d.as_secs())
        .unwrap_or(0); // Fallback to 0 if system time is somehow before UNIX_EPOCH

    Ok(Json(json!({
        "status": "healthy",
        "timestamp": timestamp,
        "version": env!("CARGO_PKG_VERSION"),
        "service": "orkee-cli"
    })))
}

pub async fn status_check() -> Result<Json<Value>> {
    let timestamp = SystemTime::now()
        .duration_since(UNIX_EPOCH)
        .map(|d| d.as_secs())
        .unwrap_or(0); // Fallback to 0 if system time is somehow before UNIX_EPOCH

    // This would contain more detailed system information in a real implementation
    Ok(Json(json!({
        "status": "healthy",
        "timestamp": timestamp,
        "version": env!("CARGO_PKG_VERSION"),
        "service": "orkee-cli",
        "uptime": timestamp, // In a real app, this would be actual uptime
        "memory": {
            "used": "unknown",
            "available": "unknown"
        },
        "connections": {
            "active": 0,
            "total": 0
        }
    })))
}

pub async fn get_csrf_token(Extension(csrf_layer): Extension<CsrfLayer>) -> Result<Json<Value>> {
    Ok(Json(json!({
        "csrf_token": csrf_layer.token()
    })))
}
